Automattic/wp-calypso
# Newsletter categories: remove feature check

## Proposed Changes

Open up newsletter categories to everyone. This PR removes the check for Atomic/Simple sites or a blog sticker. It also removes a test.

## Testing Instructions

1. Apply this PR
2. Make sure Newsletter categories show up in Calypso under Setting > Newsletters for all (Simple/Atomic/Jetpack)

## Pre-merge Checklist
